### PR TITLE
Database: clarify network attributes usage on common semconv

### PR DESCRIPTION
**Pull request by lmolkova**
## Changes

Addresses DB portion of https://github.com/open-telemetry/semantic-conventions/issues/690:

- Removes `network.transport|type` attributes from general db semantic conventions.
- Clarifies when specific systems should set `network.server.address|port` and keeps them on the cassandra, redis and elasticsearch extensions

Based on the findings from Java instrumentation libraries, following instrumentations populate network attributes:
- cassandra
- redis (Redisson, jedis, lettuce, vertx, redis)
- elastic
- opensearch

See https://github.com/open-telemetry/semantic-conventions/pull/698#issuecomment-1958869759 for more details.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* ~~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~~

### CODE REVIEW

**Review by joaopgrassi — APPROVED** [review]

**Comment by trask** [comment]
> Based on the findings from Java instrumentation libraries, following instrumentations populate network attributes:
> 
> * cassandra
> * redis (Redisson, jedis, lettuce, vertx, redis)
> * elastic
> * opensearch

do you think this is "because the instrumentation can", or because there is something about these databases that makes us include `network.peer.address` in addition to `server.address`?

what I'm trying to understand is what's the criteria for including (or excluding) `network.peer.address` in tech-specific database semconv

**Comment by lmolkova** [comment]
> > Based on the findings from Java instrumentation libraries, following instrumentations populate network attributes:
> > 
> > * cassandra
> > * redis (Redisson, jedis, lettuce, vertx, redis)
> > * elastic
> > * opensearch
> 
> do you think this is "because the instrumentation can", or because there is something about these databases that makes us include `network.peer.address` in addition to `server.address`?
> 
> what I'm trying to understand is what's the criteria for including (or excluding) `network.peer.address` in tech-specific database semconv

I can identify one good criteria - self-hosting (or hosting model where users configure/scale individual nodes).
Cassandra, Redis, and Elastic are in this camp (even in some cases there are fully-managed offerings for them).

As a user who hosts cassandra, redis, elastic, etc clusters, I'm much more interested in which specific node this operation was performed on as it helps me detect issues with that specific node. 

If I use fully managed DB (DBaaS), specific node information is probably missing (even if I have an IP address, it's probably a public one from load balancer or NAT.

**Review by alanwest — APPROVED** [review]

**Comment by lmolkova** [comment]
Update: based on the DB WG discussion on 3/8 the consensus is to keep `network.peer.*` in the general semconv with something like `recommended: if applicable` level

File: model/trace/database.yaml [hunk 1]
```diff
@@ -24,15 +22,45 @@ groups:
...
+        brief: Peer address of the database node operation was performed on.
+        requirement_level:
+          recommended: If applicable for this database system.
+        note: >
+          Semantic conventions for individual database systems SHOULD document whether `network.peer.*` attributes are applicable.
+
+          Network peer address and port are important when the application interacts with individual database nodes directly,
+          for example, when database is self-hosted.
+
+          Instrumentations for databases that don't provide self-hosted mode SHOULD NOT set this attribute.
```
Comment (trask) [inline model/trace/database.yaml:38]: "self-hosted" doesn't feel quite right to me, as it seems like peer address could be useful(?) for hosted databases too

maybe saying less here is ok?

```suggestion
          Network peer address and port are useful when the application interacts with individual database nodes directly.
```

**Review by trask — APPROVED** [review]

File: model/trace/database.yaml (hunk 1 above)
Comment (lmolkova) [reply model/trace/database.yaml]: makes sense, thank you!

**Pull request merged**